### PR TITLE
Move away from BoringSSL

### PR DESCRIPTION
**Pull request by ptoffy**
This aims to remove most if not all usages of BoringSSL and replace it with [swift-crypto](https://github.com/apple/swift-crypto/) and [swift-certificates](https://github.com/apple/swift-certificates)

### CODE REVIEW

File: Tests/JWTKitTests/X5CTests.swift [hunk 1]
```diff
@@ -155,9 +161,36 @@ final class X5CTests: XCTestCase {
...
 
-    func verify(using signer: JWTSigner) throws {
+    func verify(using _: JWTSigner) throws {
         if !cool {
             throw JWTError.claimVerificationFailure(name: "cool", reason: "not cool")
         }
     }
 }
+
+func XCTAssertNoThrowAsync<T>(
```
Comment (0xTim) [inline Tests/JWTKitTests/X5CTests.swift:171]: These should be in a separate file

File: Tests/JWTKitTests/JWTKitTests.swift [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import XCTest
 #if os(Linux)
-import FoundationNetworking
+    import FoundationNetworking
```
Comment (0xTim) [inline Tests/JWTKitTests/JWTKitTests.swift:3]: Do we need this

File: Package.swift [hunk 3]
```diff
@@ -1,32 +1,41 @@
...
     ],
     products: [
         .library(name: "JWTKit", targets: ["JWTKit"]),
-        /* This target is used only for symbol mangling. It's added and removed automatically because it emits build warnings. MANGLE_START
-        .library(name: "CJWTKitBoringSSL", type: .static, targets: ["CJWTKitBoringSSL"]),
-        MANGLE_END */
     ],
     dependencies: [
-        .package(url: "https://github.com/apple/swift-crypto.git", "2.0.0" ..< "4.0.0")
+        .package(url: "https://github.com/apple/swift-crypto.git", "2.6.0" ..< "4.0.0"),
```
Comment (0xTim) [inline Package.swift:16]: Let's just do 3.X.X for now

File: Package.swift [hunk 4]
```diff
@@ -1,32 +1,41 @@
...
+            dependencies: [
+                "JWTKit",
+            ],
+            resources: [
+                .copy("TestVectors"),
+            ]
+        ),
     ],
-     cxxLanguageStandard: .cxx11
+    cxxLanguageStandard: .cxx11
```
Comment (0xTim) [inline Package.swift:40]: I don't think we need this anymore

**Review by 0xTim — COMMENTED** [review]
The code changes in the actual business logic that go from hundreds of lines of horrible unsafe code to a few nice lines of Swift are amazing

Initial comments

File: Sources/JWTKit/Utilities/Base64URL.swift [hunk 5]
```diff
@@ -1,25 +1,46 @@
...
-        for (i, byte) in self.enumerated() {
+        for (i, byte) in enumerated() {
             switch byte {
-            case 0x2D: self[self.index(self.startIndex, offsetBy: i)] = 0x2B
-            case 0x5F: self[self.index(self.startIndex, offsetBy: i)] = 0x2F
+            case 0x2D: self[index(startIndex, offsetBy: i)] = 0x2B
+            case 0x5F: self[index(startIndex, offsetBy: i)] = 0x2F
             default: break
             }
         }
```
Comment (gwynne) [inline Sources/JWTKit/Utilities/Base64URL.swift:46]:
```swift
        for idx in self.indices {
            switch self[idx] {
            case 0x2d/* - */: self[idx] = 0x2b/* + */
            case 0x5f/* _ */: self[idx] = 0x2f/* / */
            default: break
            }
        }
```
Not only more readable but HUGELY more efficient...

File: Sources/JWTKit/Utilities/Base64URL.swift [hunk 6]
```diff
@@ -36,8 +57,8 @@ private extension Data {
     mutating func base64URLEscape() {
         for (i, byte) in enumerated() {
             switch byte {
-            case 0x2B: self[self.index(self.startIndex, offsetBy: i)] = 0x2D
-            case 0x2F: self[self.index(self.startIndex, offsetBy: i)] = 0x5F
+            case 0x2B: self[index(startIndex, offsetBy: i)] = 0x2D
+            case 0x2F: self[index(startIndex, offsetBy: i)] = 0x5F
             default: break
             }
         }
```
Comment (gwynne) [inline Sources/JWTKit/Utilities/Base64URL.swift:64]:
```swift
        for idx in self.indices {
            switch self[idx] {
            case 0x2b/* + */: self[idx] = 0x2d/* - */
            case 0x2f/* / */: self[idx] = 0x5f/* _ */
            default: break
            }
        }
```
Not only more readable but HUGELY more efficient...

**Review by gwynne — CHANGES_REQUESTED** [review]
Bunches of nits all hither and thither

**Review by 0xTim — APPROVED** [review]
There's probably a bit more work we can do on docs, like ensuring all types are linked, everything is public, a basic usage index page etc etc but that can all be done later.

BoringSSL removed! 🥳

**Pull request merged**